internal/dag/cache_test.go

```diff
@@ -36,6 +36,7 @@ import (
 	gatewayapi_v1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
 )
 
+// TODO(KauzClay): test that lookup peervalidation works
```

Comment:
Would be good to add a few test cases to cover the new logic in `LookupUpstreamValidation` if possible.